recipe-server/client/control/tests/components/recipes/test_GenericFormContainer.js

```diff
@@ -65,24 +65,23 @@ describe('<GenericFormContainer>', () => {
     expect(failed).toBe(true);
   });
 
-  it('should handle the formAction succeeding', () => {
+  it('should handle the formAction succeeding', async () => {
     const CustomEl = ({ onSubmit }) => <div onClick={onSubmit}>Test</div>;
-    let success = false;
+
+    const success = new Promise(r => r(true));
```

Comment:
This creates an immediately resolved promise, which means that the test below will always pass. A better way to do this would be 

```js
let resolve;
const success = new Promise(r => resolve = r);
onSuccess(() => { resolve(true); });
expect(await success).toBe(true);
```

I recommend making this test fail (maybe by changing the code, or maybe by removing the contents of the onSuccess handler) so you can be sure the solution you have is a valid test, instead of one that passes unconditionally.